VAGOV-1649 Press Release Detail Page

aurora-a-k-a-lightning opened the pull request:
## Description
Add the Press Release Detail Page GraphQL query and liquid template

## Testing done
Locally with staging data

## Screenshots
![localhost_3001_drupal_pittsburgh-health-care_press-releases_fayette-county-veterans-getting-larger-more-modern-va-outpatient-clinic_index html](https://user-images.githubusercontent.com/19178435/53435550-ebd15200-39ad-11e9-83e9-9b87a985ae62.png)


## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

aurora-a-k-a-lightning committed:
add press release page graphql query

aurora-a-k-a-lightning committed:
rename fragment to pressReleasePage

aurora-a-k-a-lightning committed:
press release page template

aurora-a-k-a-lightning committed:
add video download media type for press release

aurora-a-k-a-lightning committed:
Merge branch 'master' of https://github.com/department-of-veterans-affairs/vets-website into vagov-1649

aurora-a-k-a-lightning committed:
update press release styling

aurora-a-k-a-lightning committed:
Merge branch 'master' of https://github.com/department-of-veterans-affairs/vets-website into vagov-1649

ahay-agile6 commented on src/site/filters/liquid.js:
```diff
@@ -10,6 +10,8 @@ module.exports = function registerFilters() {
   liquid.filters.humanizeTimestamp = dt =>
     moment.unix(dt).format('MMMM D, YYYY');
 
+  liquid.filters.formatDate = (dt, format) => moment(dt).format(format);
```
i needed to add this because we do not have a date filter that takes any date string that is not 'YYYY-MM-DD' and a format - we're usually dealing with unix timestamps or a date string that is 'YYYY-MM-DD'.

aurora-a-k-a-lightning committed:
small ui updates

aurora-a-k-a-lightning committed:
Merge branch 'master' of https://github.com/department-of-veterans-affairs/vets-website into vagov-1649

ncksllvn commented on src/site/layouts/press_release.drupal.liquid:
```diff
@@ -0,0 +1,153 @@
+                        </div>
+                    {% endfor %}
+                </section>
+                <section class="vads-u-margin-bottom--5">###</section>
```
Should the `###` be here?

ahay-agile6 replied:
lol, it was in the mock up... seemed very important to them XD

ncksllvn commented on src/site/layouts/press_release.drupal.liquid:
```diff
@@ -0,0 +1,153 @@
+                            {% endif %}
+                            {% if a.entityBundle === 'video' %}
+                                <a href="{{ a.fieldMediaVideoEmbedField }}" target="_blank">{{ a.name }}</a>
+                            {% endif %}
```
Just preference, but seems like a [case](https://help.shopify.com/en/themes/liquid/tags/control-flow-tags#case-when) statement could be nice here. 

I'm also wondering if this should render as an unordered list? Or if there are instances where there are no downloads, should we check for `empty` before rendering this section?

ahay-agile6 replied:
ok, can do

ncksllvn commented on src/site/layouts/press_release.drupal.liquid:
```diff
@@ -0,0 +1,153 @@
+                    <p class="vads-u-margin-bottom--1">PRESS RELEASE</p>
+                    <p class="vads-u-font-weight--bold vads-u-margin-bottom--4 vads-u-margin-top--0">{{ fieldReleaseDate.value | formatDate: 'MMMM D, YYYY' }}</p>
+                    <div>
+                        <a href="#" class="vads-u-margin-right--4" onclick="window.print(); return false;"><i class="fa fa-print"></i> Print</a>
```
Since there isn't any navigation for print, I think this should be a button, and adding the `usa-button-unstyled` class will remove the button look.

ahay-agile6 replied:
ok

ahay-agile6 replied:
`usa-button-unstyled` is not working because it is being overwritten by `button, [type=button]` in the css. this is the result: 

browser:

![screen shot 2019-02-26 at 3 30 13 pm](https://user-images.githubusercontent.com/19178435/53454270-81370b00-39db-11e9-8417-193c135fba03.png)

styles:

<img width="335" alt="screen shot 2019-02-26 at 3 30 21 pm" src="https://user-images.githubusercontent.com/19178435/53454271-8300ce80-39db-11e9-8d71-c29b0064d5f6.png">

jbalboni replied:
There's a class called `va-button-link` that'll set a button to be styled as a link.

ncksllvn replied:
Ah that's what it was. Thanks @jbalboni!

ahay-agile6 replied:
ok! 'tis updated and looks wunderbar. thank you @jbalboni 😄

ncksllvn commented:
A few small suggestions, but looks great otherwise!

aurora-a-k-a-lightning committed:
code review items

- case statement for download assets
- check if the lists are empty before rendering them

aurora-a-k-a-lightning committed:
remove sidebar nav markup for now

jbalboni commented on src/site/layouts/press_release.drupal.liquid:
```diff
@@ -0,0 +1,161 @@
+                                        {% when a.entityBundle === 'image' %}
+                                            <a href="{{ a.image.url }}" download>{{ a.name }}</a>
+                                        {% when a.entityBundle === 'video' %}
+                                            <a href="{{ a.fieldMediaVideoEmbedField }}" target="_blank">{{ a.name }}</a>
```
You can remove the `target` attribute here, it'll get added if necessary based on the url in the href.

ahay-agile6 replied:
ok, removing

aurora-a-k-a-lightning committed:
print link -> print button

aurora-a-k-a-lightning committed:
remove target attribute from video link

aurora-a-k-a-lightning committed:
Merge branch 'master' of https://github.com/department-of-veterans-affairs/vets-website into vagov-1649
